python/semantic_kernel/connectors/memory/weaviate/utils.py

```diff
@@ -0,0 +1,290 @@
+    """Convert vector store data fields to Weaviate properties.
+
+    Args:
+        data_model_definition (VectorStoreRecordDefinition): The data model definition.
+
+    Returns:
+        list[Property]: The Weaviate properties.
+    """
+    properties: list[Property] = []
+
+    for field in data_model_definition.fields.values():
+        if isinstance(field, VectorStoreRecordDataField):
+            properties.append(
+                Property(
+                    name=field.name,
+                    data_type=TYPE_MAPPER_DATA[field.property_type or "default"],
+                    index_filterable=field.is_filterable,
+                    index_full_text=field.is_full_text_searchable,
+                )
+            )
+
+    return properties
+
+
+def data_model_definition_to_weaviate_named_vectors(
+    data_model_definition: VectorStoreRecordDefinition,
+) -> list[_NamedVectorConfigCreate]:
+    """Convert vector store vector fields to Weaviate named vectors.
+
+    Args:
+        data_model_definition (VectorStoreRecordDefinition): The data model definition.
+
+    Returns:
+        list[_NamedVectorConfigCreate]: The Weaviate named vectors.
+    """
+    named_vectors: list[_NamedVectorConfigCreate] = []
+
+    for vector in data_model_definition.fields.values():
+        if isinstance(vector, VectorStoreRecordVectorField):
+            named_vectors.append(
+                Configure.NamedVectors.none(
+                    name=vector.name,
+                    vector_index_config=to_weaviate_vector_index_config(vector),
+                )
+            )
+
+
+def to_weaviate_vector_index_config(vector: VectorStoreRecordVectorField) -> _VectorIndexConfigCreate:
+    """Convert a vector field to a Weaviate vector index configuration.
+
+    Args:
+        vector (VectorStoreRecordVectorField): The vector field.
+
+    Returns:
+        The Weaviate vector index configuration.
+    """
+    if vector.index_kind == IndexKind.HNSW:
+        return Configure.VectorIndex.hnsw(
+            distance_metric=to_weaviate_vector_distance(vector.distance_function),
+        )
+    if vector.index_kind == IndexKind.FLAT:
+        return Configure.VectorIndex.flat(
+            distance_metric=to_weaviate_vector_distance(vector.distance_function),
+        )
+
+    return Configure.VectorIndex.none()
+
+
+def to_weaviate_vector_distance(distance_function: DistanceFunction | None) -> str | None:
+    """Convert a distance function to a Weaviate vector distance metric.
+
+    Args:
+        distance_function (DistanceFunction | None): The distance function.
+
+    Returns:
+        str: The Weaviate vector distance metric name.
+    """
+    match distance_function:
+        case DistanceFunction.COSINE:
+            return "cosine"
+        case DistanceFunction.DOT_PROD:
+            return "dot"
+        case DistanceFunction.EUCLIDEAN:
+            return "l2-squared"
+        case DistanceFunction.MANHATTAN:
+            return "manhattan"
```

Comment:
There is also `hamming` distance function: https://weaviate.io/developers/weaviate/config-refs/distances#available-distance-metrics